### ISSUE TITLE
[Crash] Random Crash During IDE Use

### ISSUE THREAD

**Issue by ESS-BrianValdillez** [beefytech/Beef#1826]
I get an intermittent but sometimes pretty consistent crash during normal IDE usage (usually during typing). 

```
Executable: C:\Program Files\BeefLang\bin\BeefIDE.exe

File Version: 0.43.5 (Nightly 04/03/2023)

Product Version: 893beb432f408d6ace305089eb65bede8d1f8cf6


Crash minidump saved as C:\Program Files\BeefLang\bin\CrashDumps\BeefIDE_20230404_160135.dmp
Exception: Access Violation (code 0xc0000005) at address 1de'f0a38dcf in thread 3064
Logical Address: 0001:'00227dcf

f'bfd4e820 1de'f0a38dcf IDEHelper64.dll BfpProcess_IsRemoteMachine+D8EAF
    Params: 1de'81fd4710 'ffffffff 9'00000001 '00000000

f'bfd4f6d0 1de'f0a4847c IDEHelper64.dll BfpProcess_IsRemoteMachine+E855C
    Params: 101'fffd6200 1de'963f9780 1de'8b7cb000 '00000000

f'bfd4fab0 1de'f0abc617 IDEHelper64.dll BfParser_GetIndexAtLine+50CD7
    Params: 1de'81e9c840 1de'88b65700 '00000000 '00000002

f'bfd4fb40 1de'f0ad2905 IDEHelper64.dll BfCompiler_ClassifySource+145
    Params: 1de'804a504c f'bfd4fc10 1de'8a3ed730 1de'820315e0

f'bfd4fd40 7ff7'18182c3e 0001:'00221c3e BeefIDE.exe
    Params: 4'00000000 '00000000 '00000013 '00000003

f'bfd4fd70 7ff7'1819b876 0001:'0023a876 BeefIDE.exe
    Params: 1de'f06cf7e0 'ffffffff 1de'f06e3b70 '00000000

f'bfd4fdd0 7ff7'18093d07 0001:'00132d07 BeefIDE.exe
    Params: 1de'f06e3b70 1de'8eca7f40 1de'f06e3b70 1de'8eca7b80

f'bfd4fe90 7ff7'18236d3c 0001:'002d5d3c BeefIDE.exe
    Params: 1de'8214a5a0 '00003064 1de'8214a5a0 1de'823f1a40

f'bfd4fee0 7ff7'18237148 0001:'002d6148 BeefIDE.exe
    Params: 1de'8214a320 '00000000 1de'8214a320 '00000e14

f'bfd4ff10 7ffe'dd846a3b Beef042RT64.dll int bf::System::Threading::Thread::GetThreadStateNative(void)+CB
    Params: 1de'8214a5a0 '00000000 '00000000 '00000000

f'bfd4ff40 7fff'12667614 KERNEL32.DLL BaseThreadInitThunk+14
    Params: '00000000 '00000000 '00000000 '00000000

f'bfd4ffc0 7fff'140e26a1 ntdll.dll RtlUserThreadStart+21
    Params: '00000000 '00000000 '00000000 '00000000


RAX:f'bfd4efc0 RBX:1de'81e77280 RCX:'00000000 RDX:1de'8cce0b30 RSI:'00000000 RDI:'00000000
R8:1de'f081001c R9:'00000010 R10:1de'f0810000 R11:'00000000
R12:1de'81fd4710 R13:'00000000 R14:f'bfd4e8c1 R15:1de'88546050
EIP:1de'f0a38dcf ESP:f'bfd4e1e0 EBP:f'bfd4e2e0
CS:0033 SS:002B DS:002B ES:002B FS:0053 GS:002B
Flags:'00010202
```

[BeefyCrash_20230404.zip](https://github.com/beefytech/Beef/files/11153857/BeefyCrash_20230404.zip)

**Comment by bfiete** [comment]
It's almost impossible to determine root causes of bugs without a repro, but I think this will probably fix the crash: 0738fa8ce99fa71683d92d63d2b8870e740ee421

**Comment by ESS-BrianValdillez** [comment]
Thanks for taking a look, Beefy! Once we update our IDE again, I'll keep an eye out to see if it returns. 👍
